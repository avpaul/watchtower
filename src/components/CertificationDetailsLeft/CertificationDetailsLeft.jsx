import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { altDate } from '../../utils/formatDate';
import andelanEmailToName from '../../utils/formatAndelanUserEmail';

const CertificationDetailsLeft = ({ certificationInfo, vacancyInfo }) => {
  const vacancyDetails = vacancyInfo.vacancy_details || {};
  const renderTop = () => {
    const exclusive = certificationInfo.exclusive ? 'Exclusive' : 'Inclusive';
    return (
      <Fragment>
        <span className="projectType" id="projectType">
          {exclusive.toUpperCase()}
        </span>
        <p className="projectTitle">{certificationInfo.name}</p>
      </Fragment>
    );
  };

  return (
    <div className="project-container">
      <div className="project-overview">{renderTop()}</div>
      <div className="roleDetails">
        <p className="">
          {`Available Slot${vacancyInfo.available_slots > 1 ? 's : ' : ':'}`}
          <span id="availableSlots">{`${vacancyInfo.available_slots}`}</span>
        </p>
        <p>
          Start Date: <span>{` ${altDate(vacancyDetails.start_date)}`}</span>
        </p>
      </div>
      <div className="projectManager">
        <p className="headingText">Team Manager</p>
        <div className="projectManager__details">
          <img
            src="https://lorempixel.com/100/100/people/?97143"
            alt="project manager"
          />
          <div>
            <p className="managerName">
              {andelanEmailToName(vacancyDetails.requester)}
            </p>
            <p className="managerEmail">{vacancyDetails.requester}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

CertificationDetailsLeft.defaultProps = {
  certificationInfo: {},
  vacancyInfo: {}
};

CertificationDetailsLeft.propTypes = {
  certificationInfo: PropTypes.shape(),
  vacancyInfo: PropTypes.shape()
};

export default CertificationDetailsLeft;
