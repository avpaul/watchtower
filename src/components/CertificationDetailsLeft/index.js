import CertificationDetailsLeft from './CertificationDetailsLeft';

export default CertificationDetailsLeft;
