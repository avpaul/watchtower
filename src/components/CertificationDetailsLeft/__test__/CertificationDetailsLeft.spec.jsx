import React from 'react';
import { shallow } from 'enzyme';
import CertificationDetailsLeft from '../CertificationDetailsLeft';

describe('Test the certification page', () => {
  const props = {
    certificationInfo: {
      exclusive: false,
      name: 'Certification Name'
    },
    vacancyInfo: {
      vacancy_details: {},
      available_slots: 10
    }
  };

  const setUp = (propsOverride = {}) =>
    shallow(<CertificationDetailsLeft {...props} {...propsOverride} />);

  let propsOverride;

  it('should render without errors', () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });

  it('should render without errors when vacancy details are not provided', () => {
    propsOverride = props;
    delete propsOverride.vacancyInfo.vacancy_details;
    const component = setUp(propsOverride);
    expect(component).toMatchSnapshot();
  });

  it('should render type as exclusive when exclusive is true', () => {
    propsOverride = {
      certificationInfo: {
        exclusive: true
      }
    };
    const component = setUp(propsOverride);
    expect(component.find('#projectType').text()).toEqual('EXCLUSIVE');
  });

  it('should render without errors when vacancy slot is one', () => {
    propsOverride = {
      vacancyInfo: {
        available_slots: 1
      }
    };
    const component = setUp(propsOverride);
    expect(component.find('#availableSlots').text()).toEqual('1');
  });
});
