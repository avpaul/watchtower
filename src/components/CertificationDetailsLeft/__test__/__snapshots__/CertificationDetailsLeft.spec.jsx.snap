// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the certification page should render without errors 1`] = `
<div
  className="projectContainer"
>
  <div
    className="projectOverview"
  >
    <span
      className="projectType"
      id="projectType"
    >
      INCLUSIVE
    </span>
    <p
      className="projectTitle"
    >
      Certification Name
    </p>
  </div>
  <div
    className="roleDetails"
  >
    <p
      className=""
    >
      Available Slots : 
      <span
        id="availableSlots"
      >
        10
      </span>
    </p>
    <p>
      Start Date: 
      <span>
         NaN/NaN/NaN
      </span>
    </p>
  </div>
  <div
    className="projectManager"
  >
    <p
      className="headingText"
    >
      Team Manager
    </p>
    <div
      className="projectManager__details"
    >
      <img
        alt="project manager"
        src="https://lorempixel.com/100/100/people/?97143"
      />
      <div>
        <p
          className="managerName"
        />
        <p
          className="managerEmail"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Test the certification page should render without errors when vacancy details are not provided 1`] = `
<div
  className="projectContainer"
>
  <div
    className="projectOverview"
  >
    <span
      className="projectType"
      id="projectType"
    >
      INCLUSIVE
    </span>
    <p
      className="projectTitle"
    >
      Certification Name
    </p>
  </div>
  <div
    className="roleDetails"
  >
    <p
      className=""
    >
      Available Slots : 
      <span
        id="availableSlots"
      >
        10
      </span>
    </p>
    <p>
      Start Date: 
      <span>
         NaN/NaN/NaN
      </span>
    </p>
  </div>
  <div
    className="projectManager"
  >
    <p
      className="headingText"
    >
      Team Manager
    </p>
    <div
      className="projectManager__details"
    >
      <img
        alt="project manager"
        src="https://lorempixel.com/100/100/people/?97143"
      />
      <div>
        <p
          className="managerName"
        />
        <p
          className="managerEmail"
        />
      </div>
    </div>
  </div>
</div>
`;
